planning/behavior_path_planner/src/utilities.cpp

```diff
@@ -26,6 +26,20 @@
 
 namespace drivable_area_utils
 {
+template <class T>
+size_t findNearestSegmentIndex(
```

Comment:
memo: this implementation differs from the `motion_utils::findFirstNearestSegmentIndexWithSoftConstraints` since the purpose is not to find the first one.